test/any/web_catalog/api-extractor-test.es6.js

```diff
@@ -0,0 +1,331 @@
+          'arguments': 6,
+          'length': 6,
+          '+toString+': 6,
+        },
+      },
+      types,
+      'root': 10000,
+    });
+    it('obtain Function build-in properties correctly', function() {
+      extractor.getFunctionBuildInProperties_(og);
+      expect(extractor.buildInFunctionProperties.sort())
+        .toEqual(['prototype', 'caller', 'arguments', 'length', 'toString']
+        .sort());
+    });
+    it('clean up string for properties', function() {
+      extractor.getFunctionBuildInProperties_(og);
+      expect(extractor.buildInFunctionProperties).toContain('toString');
+    });
+  });
+
+  describe('Method getObjectBuildInProperties', function() {
+    let og = objectGraph.fromJSON({
+      'data': {
+        '10000': {
+          'Object': 10001,
+        },
+        '10001': {
+          'prototype': 10002,
+        },
+        '10002': {
+          '+toString+': 3,
+          '+valueOf+': 6,
+          '+constructor': 6,
+        },
+      },
+      types,
+      'root': 10000,
+    });
+    it('obtain cleaned up Object build-in properties correctly', function() {
+      extractor.getObjectBuildInProperties_(og);
+      expect(extractor.buildInObjectProperties.sort())
+        .toEqual(['toString', 'valueOf', 'constructor']
+        .sort());
+    });
+  });
+
+  describe('Method getClassName', function() {
+    it('obtain class name of object\'s proto is its class', function() {
+      let og = objectGraph.fromJSON({
+        'data': {
+          '10010': {
+            'instanceProperty': 3,
+          },
+          '10020': {
+            '+constructor+': 10030,
+          },
+        },
+        'protos': {
+          '10010': '10020',
+        },
+        'functions': {
+          '10030': 'class',
+        },
+        types,
+      });
+      expect(extractor.getClassName_(10010, og)).toEqual('class');
+    });
+    it('obtain class name if it is the class is deep in the prototype chain',
+    function() {
+      let og = objectGraph.fromJSON({
+        'data': {
+          '10010': {
+            'instanceProperty': 3,
+          },
+          '10020': {
+          },
+          '10030': {
+          },
+          '10040': {
+            '+constructor+': 10050,
+          },
+        },
+        'protos': {
+          '10010': '10020',
+          '10020': '10030',
+          '10030': '10040',
+        },
+        'functions': {
+          '10020': 'noClass',
+          '10030': 'noClass',
+          '10040': 'classPrototype',
+          '10050': 'class',
+        },
+        types,
+      });
+      expect(extractor.getClassName_(10010, og)).toEqual('class');
+    });
+      it('return null if cannot find class', function() {
+      let og = objectGraph.fromJSON({
+        'data': {
+          '10010': {
+            'instanceProperty': 3,
+          },
+          '10020': {
+          },
+          '10030': {
+          },
+        },
+        'protos': {
+          '10010': '10020',
```

Comment:
Do we need multiple layers of protos for this test? If that's what you're testing, please add it to the `it(...)` description. If not, let's keep the simplest possible graph for what we intend to test.